core/src/main/java/fr/sncf/osrd/conflicts/SpacingResourceGenerator.kt

```diff
@@ -110,12 +105,6 @@ class SpacingRequirementAutomaton(
         pendingRequirements.add(req)
     }
 
-    private fun getSignalProtectedZone(signal: PathSignal): Int {
-        // the signal protects all zone inside the block
-        // if a signal is at a block boundary,
-        return incrementalPath.getBlockEndZone(signal.minBlockPathIndex)
```

Comment:
I believe this method behaves properly. There are two main cases:
 - either the signal is a block delimiter, like a BAL3 signal. in this case, minBlockPathIndex returns the index of block which ends at this signal. Then getBlockEndZone returns the index of the zone after the block.
 - if the signal does not delimit a block, like a BAPR distant signal, minBlockPathSignal is the current block, and getBlockEndZone is the index of the next zone after the block

I believe the documentation of getSignalProtectedZone should be improved, and the documentation of getBlockEndZone should make clear it returns an excluded bound